packages/jsActions/nanoflow-actions-native/src/external/DownloadFile.ts

```diff
@@ -0,0 +1,47 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import fetchBlob from "rn-fetch-blob";
+
+// BEGIN EXTRA CODE
+function formatPath(fileURL: string, filePath: string, fileName: string): string {
+    if (filePath && fileName) {
+        return `/${filePath}/${fileName}`;
+    }
+    if (fileName) {
+        return `/${fileName}`;
+    }
+    const urlFileName = fileURL.substring(fileURL.lastIndexOf("/") + 1);
+    if (filePath) {
+        return `/${filePath}/${urlFileName}`;
+    }
+    return `/${urlFileName}`;
+}
+// END EXTRA CODE
+
+/**
+ * @param {string} fileURL - The file Url is required
+ * @param {string} filePath - optional
+ * @param {string} fileName - optional - should be with extention
+ * @returns {Promise.<boolean>}
+ */
+export async function DownloadFile(fileURL: string, filePath: string, fileName: string): Promise<boolean> {
+    // BEGIN USER CODE
+    const dirs = fetchBlob.fs.dirs;
```

Comment:
I think you should make all properties optional and check if they are required.

Something like this

```suggestion
export async function DownloadFile(fileURL?: string, filePath?: string, fileName?: string): Promise<boolean> {
    // BEGIN USER CODE
    if (!fileURL) {
        return Promise.reject(new Error("Input parameter 'File URL' is required"));
    }
```

Reply:
done